api/v1beta1/azurecluster_webhook.go

```diff
@@ -168,7 +168,7 @@ func (c *AzureCluster) validateSubnetUpdate(old *AzureCluster) field.ErrorList {
 						c.Spec.NetworkSpec.Subnets[i].RouteTable.Name, "field is immutable"),
 				)
 			}
-			if subnet.NatGateway.Name != oldSubnet.NatGateway.Name {
+			if (subnet.NatGateway.Name != oldSubnet.NatGateway.Name) && (oldSubnet.NatGateway.Name != "") {
```

Comment:
nit: we could use `webhookutils.ValidateZeroTransition` for this 

but I see we're not doing this consistently in this current file so also good with merging this as is for now